source/docs/yearly-overview/yearly-changelog.rst

```diff
@@ -0,0 +1,138 @@
+
+- Added ``EllipticalRegionConstraint``, ``RectangularRegionConstraint``, and ``MaxVelocityConstraint`` to allow constraining trajectory velocity in a certain region of the field.
+
+- Added equals() operator to the ``Trajectory`` class to compare two or more trajectories.
+
+- Added zero-arg constructor to the ``Trajectory`` class in Java that creates an empty trajectory.
+
+- Added a special exception to catch trajectory constraint misbehavior. This notifies users when user-defined constraints are misbehaving (i.e. min acceleration is greater than max acceleration).
+
+- Added a ``getRotation2d()`` method to the ``Gyro`` interface. This method automatically takes care of converting from gyro conventions to geometry conventions.
+
+- Added angular acceleration units for C++ teams. These are available in the ``<units/angular_acceleration.h>`` header.
+
+- Added X and Y component getters in ``Pose2d`` - ``getX()`` and ``getY()`` in Java, ``X()`` and ``Y()`` in C++.
+
+- Added implicit conversion from ``degree_t`` to ``Rotation2d`` in C++. This allows teams to use a degree value (i.e. ``47_deg``) wherever a ``Rotation2d`` is required.
+
+- Fixed some spline generation bugs for advanced users who were using control vectors directly.
+
+- Fixed theta controller continuous input in swerve examples. This fixes the behavior where the shortest path is not used during drivetrain rotation.
+
+- Deprecated ``units.h``, use individual units headers instead.
+
+- Added support for scheduling functions more often than the robot loop via addPeriodic() in TimedRobot. Previously, teams had to make a Notifier to run feedback controllers more often than the TimedRobot loop period of 20ms (running TimedRobot more often than this is not advised). Now, users can run feedback controllers more often than the main robot loop, but synchronously with the TimedRobot periodic functions so there aren't any thread safety issues.
+
+Breaking Changes
+^^^^^^^^^^^^^^^^
+
+- ``curvature_t`` moved from ``frc`` to ``units`` namespace (C++)
+
+- Trajectory constraint methods are now ``const`` in C++. Teams defining their own custom constraints should mark the ``MaxVelocity()`` and ``MinMaxAcceleration()`` methods as ``const``.
+
+Simulation
+----------
+
+- Add joystick simulation support.
+- Added Mechanism2D for simulating mechanisms.
+
+Shuffleboard
+------------
+
+- Number Slider now displays the text value
+- Graphing Widget now uses ChartFX, a high performance graphing library
+- Fix decimal digit formatting with large numbers
+- Size and position can now be set separately in the Shuffleboard API
+
+SmartDashboard
+--------------
+
+- Host IP can be specified in configuration.
+
+
+FRC Raspberry Pi Image
+----------------------
+
+Any changes this summer?
+
+PathWeaver
+----------
+
+- Added support for reversed splines
+
+OutlineViewer
+-------------
+
+Any changes this summer?
+
+GradleRIO
+---------
+
+Any changes this summer?
```

Comment:
```suggestion
- Added a `vendordep` task for downloading vendor JSONs or fetching them from the user `wpilib` folder
- Added a `gradlerio.vendordep.folder.path` property to set a non-default location for the vendor JSON folder
- Renamed the `wpi` task (that prints current versions of WPILib and tools) to `wpiVersions`
```
This is simply from a quick look at the recently-merged GradleRIO PRs (some of them are mine), maybe @ThadHouse has some more additions.  wpilibsuite/GradleRIO#426 also exists, but I'm not sure how to sum up the changes
There's also the wpimath split but I don't think that it's a GradleRIO change.